autogpts/autogpt/autogpt/core/resource/model_providers/openai.py

```diff
@@ -406,83 +411,90 @@ async def create_chat_completion(
     ) -> ChatModelResponse[_T]:
```

Comment:
How about passing `list[Command]` as `functions` and letting llm convert them to whatever it needs

Reply:
That's out of scope for this module. Every command is a function, but not every function is a command.